BuildMacOSUniversalBinary.py

```diff
@@ -0,0 +1,278 @@
+    2) Files that exist in both trees and are identical are copied over
+       unmodified
+    3) Files that exist in both trees and are non-identical are lipo'd
+    4) Symlinks are created in the destination tree to mirror the hierarchy in
+       the source trees
+    """
+
+    # loop over all files in src0
+    for newpath0 in glob.glob(src0+"/*"):
+        filename = os.path.basename(newpath0)
+        newpath1 = os.path.join(src1, filename)
+        new_dst_path = os.path.join(dst, filename)
+        if os.path.islink(newpath0):
+            # symlinks will be fixed after files are resolved
+            continue
+
+        if not os.path.exists(newpath1):
+            # copy files that don't exist in path1
+            shutil.copy(newpath0, new_dst_path)
+            continue
+
+        if os.path.isdir(newpath1):
+            os.mkdir(new_dst_path)
+            # recurse into directories
+            recursiveMergeBinaries(newpath0, newpath1, new_dst_path)
+            continue
+
+        if filecmp.cmp(newpath0, newpath1):
+            # copy files that are the same
+            shutil.copy(newpath0, new_dst_path)
+        else:
+            # lipo together files that are different
+            lipo(newpath0, newpath1, new_dst_path)
+
+    # loop over files in src1 and copy missing things over to dst
+    for newpath1 in glob.glob(src1+"/*"):
+        filename = os.path.basename(newpath0)
+        newpath0 = os.path.join(src0, filename)
+        new_dst_path = os.path.join(dst, filename)
+        if not os.path.exists(newpath0) and not os.path.islink(newpath1):
+            shutil.copytree(newpath1, new_dst_path)
+
+    # fix up symlinks for path0
+    for newpath0 in glob.glob(src0+"/*"):
+        filename = os.path.basename(newpath0)
+        new_dst_path = os.path.join(dst, filename)
+        if os.path.islink(newpath0):
+            relative_path = os.path.relpath(os.path.realpath(newpath0), src0)
+            os.symlink(relative_path, new_dst_path)
+    # fix up symlinks for path1
+    for newpath1 in glob.glob(src1+"/*"):
+        filename = os.path.basename(newpath1)
+        new_dst_path = os.path.join(dst, filename)
+        newpath0 = os.path.join(src0, filename)
+        if os.path.islink(newpath1) and not os.path.exists(newpath0):
+            relative_path = os.path.relpath(os.path.realpath(newpath1), src1)
+            os.symlink(relative_path, new_dst_path)
+
+
+def build(config):
+    """
+    Builds the project with the parameters specified in config.
+    """
+
+    print("Building config:")
+    print(json.dumps(config, indent=4))
+
+    dst_app = config["dst_app"]
+    # Configure and build single architecture builds for each architecture
+    for arch in ARCHITECTURES:
+        # Create build directory for architecture
+        if not os.path.exists(arch):
+            os.mkdir(arch)
+        # Setup environment variables for build
+        envs = os.environ.copy()
```

Comment:
AFAIK, it should be safe to name it `env`.

Reply:
Sure, I can rename it to that, it doesn't cause any problems.